crates/pera-types/src/messages_dwallet_mpc.rs

```diff
@@ -15,18 +15,17 @@ pub enum MPCRound {
     /// The first round of the Presign protocol.
     /// Contains the `ObjectId` of the dWallet object,
     /// the DKG decentralized output, and the batch session ID.
-    PresignFirst(ObjectID, Vec<u8>, ObjectID),
+    PresignFirst(ObjectID, MPCPublicOutput, ObjectID),
     /// The second round of the Presign protocol.
     /// Contains the `ObjectId` of the dWallet object,
     /// the Presign first round output, and the batch session ID.
-    PresignSecond(ObjectID, Vec<u8>, ObjectID),
+    PresignSecond(ObjectID, MPCPublicOutput, ObjectID),
     /// The first and only round of the Sign protocol.
-    /// Contains the `PartyID` associated with the decryption share,
-    /// the `ObjectID` of the batched sign session,
-    /// and the hashed message being signed.
-    Sign(ObjectID, Vec<u8>),
+    /// Contains the `ObjectID` of the batched sign session,
+    /// and the hashed messages being signed.
+    Sign(ObjectID, MPCPublicOutput),
     /// A batched sign session, contains the list of messages that are being signed.
-    BatchedSign(Vec<Vec<u8>>),
+    BatchedSign(Vec<MPCMessage>),
```

Comment:
I think MPCMessage is not apt as this is a message the user provided to sign on, not a message sent between validators

Reply:
not a big fan of this "type to explain field" practice in general, we'll better have structs for each round's inner data with descriptive field names